v2/cmd/controller/main.go

```diff
@@ -86,7 +87,20 @@ func main() {
 	log.V(Status).Info("Configuration details", "config", cfg.String())
 
 	if cfg.OperatorMode.IncludesWatchers() {
-		if errs := controllers.RegisterAll(mgr, clientFactory, controllers.GetKnownStorageTypes(), makeControllerOptions(log, cfg)); errs != nil {
+		var extensions map[schema.GroupVersionKind]genruntime.ResourceExtension
+		extensions, err = controllers.GetResourceExtensions(scheme)
```

Comment:
Why not just ` extensions, err := ...` here? It seems like contortions to avoid shadowing `err` from outside the `if`, but is there a reason to do that?

Reply:
The linting pass in our CI complains bitterly about shadowing variables, even just for `err`. I've had to do this dance a couple of times myself. I understand _why_ it complains, and believe it's a good check to have. But just occasionally, it forces a bit of break-dancing.

![gif](https://media.giphy.com/media/1Mng0gXC5Tpcs/giphy.gif)

Reply:
Yeah this is to make the linter happy. Happy to do something else but it's not clear what's better than this. I could suppress the linter here (seems like if I'm doing that I should just turn it off?)

Reply:
The only case where shadowing might be acceptable is for `err` - being unable to shadow that seems a small price to ensure we catch other cases.